pilot/pkg/util/protoconv/protoconv.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protoconv
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"istio.io/pkg/log"
+)
+
+// MessageToAnyWithError converts from proto message to proto Any
+func MessageToAnyWithError(msg proto.Message) (*anypb.Any, error) {
```

Comment:
copy from pkg/netwroking to avoid import cycle

Reply:
this function and MessageToAny are in `pilot/pkg/networking` and `pilot/pkg/networking/util`.  Is there a reason these can't all use the same function?

Reply:
can also be addressed in a followup PR since the refactor is outside the scope of this PR

Reply:
I will refactor all of those to same place in a separate PR.